extensions/podman/scripts/download.ts

```diff
@@ -0,0 +1,130 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import { https } from 'follow-redirects';
+import * as fs from 'node:fs';
+import * as path from 'node:path';
+
+const tools = require('../src/podman.json');
+
+import * as hasha from 'hasha';
+
+const platform = process.platform;
+
+const MAX_DOWNLOAD_ATTEMPT = 3;
+let downloadAttempt = 0;
+
+// to make this file a module
+export {};
+
+function downloadFile(url, dest): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const file = fs.createWriteStream(dest, { flags: 'wx' });
+
+    const request = https.get(url, { headers: { 'User-Agent': 'Podman Desktop' } }, response => {
+      if (response.statusCode === 200) {
+        response.pipe(file);
+      } else {
+        file.close();
+        // eslint-disable-next-line @typescript-eslint/no-empty-function
+        fs.unlink(dest, () => {}); // Delete temp file
+        reject(`Server responded with ${response.statusCode}: ${response.statusMessage}`);
+      }
+    });
+
+    request.on('error', err => {
+      file.close();
+      // eslint-disable-next-line @typescript-eslint/no-empty-function
+      fs.unlink(dest, () => {}); // Delete temp file
+      reject(err.message);
+    });
+
+    file.on('finish', () => {
+      resolve();
+    });
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    file.on('error', err => {
+      file.close();
+      console.error(err);
+      if ((err as unknown as { code: string }).code === 'EEXIST') {
+        reject('File already exists');
+      } else {
+        // eslint-disable-next-line @typescript-eslint/no-empty-function
+        fs.unlink(dest, () => {}); // Delete temp file
+        reject(err.message);
+      }
+    });
+  });
+}
+async function downloadPackage(url, fileName): Promise<string | undefined> {
+  if (url && fileName) {
+    const destDir = path.resolve(__dirname, '..', 'assets');
+    if (!fs.existsSync(destDir)) {
+      fs.mkdirSync(destDir);
+    }
+    const destFile = path.resolve(destDir, fileName);
+    if (!fs.existsSync(destFile)) {
+      console.log(`Downloading Podman package from ${url}`);
+      await downloadFile(url, destFile);
+      console.log(`Downloaded to ${destFile}`);
+      return destFile;
+    } else {
+      console.log(`Podman package ${url} already downloaded. Exiting...`);
+      return destFile;
+    }
+  } else {
+    console.error(
+      `Can't find Podman package to download. Please look on ${path.resolve(__dirname, '..', 'podman.json')}`,
+    );
+    process.exit(42);
+  }
+}
+
+async function checkFileSha(filePath: string, shaSum: string): Promise<boolean> {
+  const sha256sum: string = await hasha.fromFile(filePath, { algorithm: 'sha256' });
```

Comment:
should switch over to sha512

Reply:
see https://github.com/containers/podman-desktop/pull/166#discussion_r887836266